src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,2857 @@
+
+        {
+            var cast = casts[0];
+            Assert.Equal("(ReadOnlySpan<T>)x", cast.ToString());
+
+            var op = (IConversionOperation)model.GetOperation(cast)!;
+            var conv = op.GetConversion();
+            Assert.Equal(ConversionKind.ImplicitSpan, conv.Kind);
+
+            model.VerifyOperationTree(cast, """
+                IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: System.ReadOnlySpan<T>) (Syntax: '(ReadOnlySpan<T>)x')
+                  Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+                  Operand:
+                    IParameterReferenceOperation: x (OperationKind.ParameterReference, Type: T[]) (Syntax: 'x')
+                """);
+        }
+
+        {
+            var cast = casts[1];
+            Assert.Equal("(ReadOnlySpan<U>)x", cast.ToString());
+
+            var op = (IConversionOperation)model.GetOperation(cast)!;
+            var conv = op.GetConversion();
+            Assert.Equal(ConversionKind.ExplicitUserDefined, conv.Kind);
+
+            model.VerifyOperationTree(cast, """
+                IConversionOperation (TryCast: False, Unchecked) (OperatorMethod: System.ReadOnlySpan<U> System.ReadOnlySpan<U>.op_Implicit(U[] array)) (OperationKind.Conversion, Type: System.ReadOnlySpan<U>) (Syntax: '(ReadOnlySpan<U>)x')
+                  Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: True) (MethodSymbol: System.ReadOnlySpan<U> System.ReadOnlySpan<U>.op_Implicit(U[] array))
+                  Operand:
+                    IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: U[], IsImplicit) (Syntax: 'x')
+                      Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: True, IsUserDefined: False) (MethodSymbol: null)
+                      Operand:
+                        IParameterReferenceOperation: x (OperationKind.ParameterReference, Type: T[]) (Syntax: 'x')
+                """);
+        }
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Implicit_UnrelatedElementType(LanguageVersion langVersion)
+    {
+        var source = """
+            class C
+            {
+                System.Span<string> M(int[] arg) => arg;
+            }
+            """;
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (3,41): error CS0029: Cannot implicitly convert type 'int[]' to 'System.Span<string>'
+            //     System.Span<string> M(int[] arg) => arg;
+            Diagnostic(ErrorCode.ERR_NoImplicitConv, "arg").WithArguments("int[]", "System.Span<string>").WithLocation(3, 41));
+    }
+
+    [Fact]
+    public void Conversion_Array_Span_Implicit_NullableAnalysis()
```

Comment:
I'm honestly confused why any of these work without warning. Shouldn't this all be an invariant scenario? Or does it fall out of array covariance?

Reply:
Let's just leave a quick prototype comment above one of these tests to revisit.